### ISSUE TITLE
hickory-dns (trust-dns) should be built with `dns-over-rustls` and `dns-over-https-rustls` features enabled

### ISSUE THREAD

**Issue by michael-berg** [NixOS/nixpkgs#358831]
## Describe the bug

The current hickory-dns (formerly trust-dns) package build does not support DNS-over-TLS (DoT) or DNS-over-HTTPS (DoH) when forwarding queries to an upstream nameserver that supports DoT or DoH.

## Steps To Reproduce

Try to run hickory-dns using the following manually-generated `hickory-dns.toml` file:
~~~
listen_addrs_ipv4 = ["0.0.0.0"]
listen_addrs_ipv6 = ["::"]
listen_port = 53
directory = "/var/lib/hickory-dns"

[[zones]]
zone = "."
zone_type = "Forward"

[[zones.stores]]
type = "forward"

[[zones.stores.name_servers]]
socket_addr = "9.9.9.9:853"
protocol = "tls"
tls_dns_name = "dns.quad9.net"
trust_negative_responses = true
~~~

1. Place the above minimal TOML file somewhere like `/etc/hickory-dns/hickory-dns.toml`
2. `/nix/store/q3qxwm2jrzlxpdwhj6f7nv9n5hi0w1jp-hickory-dns-0.25.0-alpha.3/bin/hickory-dns --config /etc/hickory-dns/hickory-dns.toml`

~~~
1732486825:INFO:hickory_dns:427:Hickory DNS 0.25.0-alpha.3 starting...
1732486825:INFO:hickory_dns:434:loading configuration from: "/etc/hickory-dns/hickory-dns.toml"
Error: failed to read config file from "/etc/hickory-dns/hickory-dns.toml": toml decode error: TOML parse error at line 33, column 1
   |
33 | [[zones.stores]]
   | ^^^^^^^^^^^^^^^^
unknown variant `tls`, expected `udp` or `tcp`
~~~

## Expected behavior

The above example should start normall, supporting `protocol = "tls"` in order to use DNS-over-TLS (DoT) with the specified upstream resolver.

## Additional context

Per the [hickory-dns README](https://github.com/hickory-dns/hickory-dns?tab=readme-ov-file#dns-over-tls-and-dns-over-https), "To enable DoT, one of the features `dns-over-native-tls`, `dns-over-openssl`, or `dns-over-rustls` must be enabled. `dns-over-https-rustls` is used for DoH."

At a minimum, the Nix package can hopefully support one of the DoT features.

Ideally, both of the features `dns-over-rustls` and `dns-over-https-rustls` can be enabled in the build for both DoT and DoH support using the more secure `rustls` implementation.

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.60, NixOS, 24.11 (Vicuna), 24.11beta708443.057f63b6dc1a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - channels(root): `"home-manager-24.11.tar.gz, nixos-24.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

## Notify maintainers

@colinsane

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by cpu** [comment]
The `dns-over-rustls`/`dns-over-https-rustls` features no long exist in 0.25/0.26, and the current derivation includes the `-ring` features for supporting DoT,DoQ,DoH2 with rustls+ring: https://github.com/NixOS/nixpkgs/blob/9dc89b1672e4f4c14ca095342488c14b41fe2f33/pkgs/by-name/hi/hickory-dns/package.nix#L23-L32

I think this issue could be closed.

**Comment by michael-berg** [comment]
Yes, with the recent versions of hickory-dns, the new `buildFeatures` in the unstable NixOS package enable the required features. The package in the stable 25.11 is still missing these features. However, the unstable version will be released as 26.05 soon, so closing this issue.
